backend/src/packages/users/user.service.ts

```diff
@@ -30,16 +42,31 @@ class UserService implements IService {
   public async create(
     payload: UserSignUpRequestDto,
   ): Promise<UserSignUpResponseDto> {
+    const saltRounds = 10;
+    const salt = await bcrypt.genSalt(saltRounds);
+    const hashedPassword = await bcrypt.hash(payload.password, salt);
+
     const user = await this.userRepository.create(
       UserEntity.initializeNew({
         email: payload.email,
-        /** @todo replace with real implementation */
-        passwordSalt: 'SALT',
-        passwordHash: 'HASH',
+        passwordSalt: salt,
+        passwordHash: hashedPassword,
       }),
     );
 
-    return user.toObject();
+    const userDetails = await this.userDetailsRepository
+      .create(
+        UserDetailsEntity.initializeNew({
+          lastName: payload.lastName,
+          firstName: payload.firstName,
+          userId: user.getId() as number,
+        }),
+      )
+      .then((data) => data.toNewObject());
+
+    const { firstName, lastName } = userDetails;
+
+    return { ...user.toObject(), firstName, lastName };
```

Comment:
All user data should be processed through user entity
```suggestion
    return user.toObject();
```